jib-core/src/main/java/com/google/cloud/tools/jib/configuration/credentials/CredentialRetriever.java

```diff
@@ -27,7 +27,9 @@
    *
    * @return the fetched credentials or {@code null} if no credentials could be fetched with this
    *     provider
+   * @throws Exception if the execution should fail instead of trying other credential retrieval
```

Comment:
Can't get what this really means. Does it mean

"the retrieval attempt this `CredentialRetriever` tried failed if you got an `Exception` (but then what "other credential methods" mean? Is `CredentialRetriever` supposed to try multiple methods in general? Even so, I guess it's an implementation detail.)

or

"if you as a caller is in the middle of trying multiple `CredentialRetriever`s and got any `Exception` from any of them, you as a caller should stop trying other retrievers and "fail" right away."

In any case, I think the consumer of the Javadoc of an interface should be the caller in general.

Reply:
Hmm, I was intending the consumer of this javadoc to be the user implementing the interface. The intention was:

"If you decide to have this throw an exception, it indicates that something went awry and execution should not continue (whereas if you decide to have this return null, it just indicates that this retrieval failed to retrieve any credentials, but execution should continue on)."

Reply:
Reworded it a bit.

Reply:
I believe Javadoc of almost every class or interface is intended for the users of them. If you want to direct implementors to something specific or certain implementation details, I'd say, e.g., "implementation of this class should be thread-safe", "note to implementors:", or such.